### DISCUSSION TITLE
Problem building my app

### DISCUSSION THREAD

**Discussion by SpaceSimon** [thebergamo/react-native-fbsdk-next#360] [Q&A] [answered]
Hello,

**With version 10.1.0**

I cannot build my app on Android (real device, tried in debug and realease). I followed the instructions:
- changed mavenLocal() to mavenCentral()
- changed `implementation 'com.facebook.android:facebook-android-sdk:[5,6)'` to `implementation 'com.facebook.android:facebook-android-sdk:latest.release'`
- added `<string name="facebook_client_token">myclienttoken</string>` in strings.xml (I already had `<string name="facebook_app_id">myappid</string>`)
- added `<meta-data android:name="com.facebook.sdk.ClientToken" android:value="@string/facebook_client_token"/>` in AndroidManifest.xml (I already had `<meta-data android:name="com.facebook.sdk.ApplicationId" android:value="@string/facebook_app_id"/>`)
- already had `<uses-permission android:name="android.permission.INTERNET"/>`

I had the old fbsdk working fine back then, so I suppose my Facebook App is still valid and the keys are working.

Here is the error in the terminal when trying to build:

```
> Configure project :app
Reading env from: .env.staging

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/myuser/Documents/myapp/node_modules/react-native-config/android/dotenv.gradle' line: 50

* What went wrong:
A problem occurred evaluating script.
> dotenv_dfjhf39zhwbeqbjvmcgaf8voi$_loadDotEnv_closure4

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':app'.
> compileSdkVersion is not specified. Please add it to build.gradle

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
```

The only part that makes sense to me is `compileSdkVersion is not specified`, but of course it is defined.

Thanks for your help

**Comment by SpaceSimon** [comment, marked as answer]
OK I have found why and you are not going to like it.

Apparently `react-native-config` keeps failing after changing 'things' in your project, even if you clean everything. You have to restart your computer: https://github.com/luggit/react-native-config/issues/355

**Reply by mikehardy** [reply]
> OK I have found why and you are not going to like it.

I don't use react-native-config so I have no personal feelings on it, as maintainer here at react-native-fbsdk-next I have no professional opinion. This is a discussion entirely focused on a package that has no relation to this package and so is in the wrong spot I think?

**Reply by SpaceSimon** [reply]
I created a discussion instead of an issue because I thought my problem was related to react-native-fbsdk-next but not only (maybe a compatibility problem with another lib).

Then I found out it was solely due to react-native-config. You are right I should post this on react-native-config and delete this one.
